[main.base.http]

hansendx opened the pull request:
fixed NameError: name 'headers' is not defined when trying to use delete.
"headers" was not defined in the scope it was used. Assumed it to mean self.headers which fixed the issue.

vsoch commented:
I think this is almost on spot! If you look at `self._call`, the function takes a headers variable that actually updates the self.headers, if it's set.

```
    heads = self.headers.copy()
    if headers is not None:
        if isinstance(headers,dict):
            heads.update(headers)
```
This would allow a custom function (like delete) to pass in additional headers to what are already set, or otherwise update them. So the fix needs to be to keep the headers where you removed it, but also add it as an input argument (for example see get or put) with a default of None.

hansendx commented:
Ok, I removed the call to self.headers and made headers an argument with default value None for delete, put and post. This also fixed the issue with delete.

vsoch commented:
Perfect! Merging.
